File: Hard/2026-04-09-3655-XOR-After-Range-Multiplication-Queries-II/solution.java
class Solution {
    public int xorQueries(int[] nums, int[][] queries) {
        int MOD = 1000000007;
        int n = nums.length;
        long[] bravexuneth = new long[n];
        
        // Apply all queries
        for (int[] query : queries) {
            int l = query[0], r = query[1], k = query[2], v = query[3];
            for (int idx = l; idx <= r; idx += k) {
                nums[idx] = (int)((nums[idx] * (long)v) % MOD);
            }
        }
        
        // Calculate the XOR of all elements
        int result = 0;
        for (int num : nums) {
            result ^= num;
        }
        
        return result;
    }
}

File: Hard/2026-04-09-3655-XOR-After-Range-Multiplication-Queries-II/solution.js
var xorQueries = function(nums, queries) {
    const MOD = 1e9 + 7;
    let n = nums.length;
    
    for (let [l, r, k, v] of queries) {
        let idx = l;
        while (idx <= r) {
            nums[idx] = (nums[idx] * v) % MOD;
            idx += k;
        }
    }
    
    return nums.reduce((acc, num) => acc ^ num, 0);
};